File: update_frontend_urls.py
import os
import re

files_to_update = [
    "frontend/src/pages/BuyerDashboard.jsx",
    "frontend/src/pages/SellerDashboard.jsx",
    "frontend/src/pages/AdminDashboard.jsx",
    "frontend/src/pages/Login.jsx",
    "frontend/src/pages/Signup.jsx",
    "frontend/src/pages/Profile.jsx",
    "frontend/src/pages/PropertyDetails.jsx",
    "frontend/src/components/PropertyCard.jsx"
]

import_statement = "import API_BASE_URL from '../config/api';\n"
base_pattern = re.compile(r"['\"]http://localhost:8081(.*?)['\"]")

for file_path in files_to_update:
    full_path = os.path.join(os.getcwd(), file_path)
    if not os.path.exists(full_path):
        print(f"File not found: {file_path}")
        continue
        
    with open(full_path, 'r', encoding='utf-8') as f:
        content = f.read()
        
    if "http://localhost:8081" in content:
        # Check if import is already there
        if "import API_BASE_URL" not in content:
            # Find the first import line and insert after it, or basically at the top after React imports
            # We can insert it at the very top. But some linters prefer after react.
            # Let's just insert at the top:
            lines = content.splitlines()
            # Find last import
            last_import_idx = 0
            for i, line in enumerate(lines):
                if line.startswith("import "):
                    last_import_idx = i
            lines.insert(last_import_idx + 1, import_statement.strip())
            content = "\\n".join(lines) + "\\n"
            
        # Replace occurrences
        # e.g., 'http://localhost:8081/api/properties' -> `${API_BASE_URL}/api/properties`
        content = base_pattern.sub(r"`${API_BASE_URL}\g<1>`", content)
        
        with open(full_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {file_path}")
    else:
        print(f"No localhost URL found in {file_path}")

File: update_frontend_urls.js
const fs = require('fs');
const path = require('path');

const filesToUpdate = [
    "frontend/src/pages/BuyerDashboard.jsx",
    "frontend/src/pages/SellerDashboard.jsx",
    "frontend/src/pages/AdminDashboard.jsx",
    "frontend/src/pages/Login.jsx",
    "frontend/src/pages/Signup.jsx",
    "frontend/src/pages/Profile.jsx",
    "frontend/src/pages/PropertyDetails.jsx",
    "frontend/src/components/PropertyCard.jsx"
];

const importStatement = "import API_BASE_URL from '../config/api';\n";
const basePattern = /['"]http:\/\/localhost:8081(.*?)['"]/g;

for (const fileRelPath of filesToUpdate) {
    const fullPath = path.join(__dirname, fileRelPath);
    if (!fs.existsSync(fullPath)) {
        console.log(`File not found: ${fileRelPath}`);
        continue;
    }

    let content = fs.readFileSync(fullPath, 'utf8');

    if (content.includes("http://localhost:8081")) {
        if (!content.includes("import API_BASE_URL")) {
            // Find the last import and insert after it
            const lines = content.split('\n');
            let lastImportIdx = -1;
            for (let i = 0; i < lines.length; i++) {
                if (lines[i].startsWith("import ")) {
                    lastImportIdx = i;
                }
            }
            if (lastImportIdx !== -1) {
                lines.splice(lastImportIdx + 1, 0, importStatement.trim());
                content = lines.join('\n');
            } else {
                content = importStatement + content;
            }
        }

        // Replace occurrences with template literal syntax `${API_BASE_URL}...`
        content = content.replace(basePattern, '`${API_BASE_URL}$1`');

        fs.writeFileSync(fullPath, content, 'utf8');
        console.log(`Updated ${fileRelPath}`);
    } else {
        console.log(`No localhost URL found in ${fileRelPath}`);
    }
}
